resources/assets/lib/components/portal.tsx

```diff
@@ -3,41 +3,33 @@
 
 import { PureComponent, ReactNode } from 'react';
 import { createPortal } from 'react-dom';
-import { nextVal } from 'utils/seq';
 
 interface Props {
   children: ReactNode;
 }
 
-export class Portal extends PureComponent<Props> {
-  private readonly container: HTMLElement;
-  private readonly eventId = `portal-${nextVal()}`;
-
-  constructor(props: Props) {
-    super(props);
-
-    this.container = document.createElement('div');
-  }
-
-  addPortal = () => document.body.appendChild(this.container);
+export default class Portal extends PureComponent<Props> {
+  private readonly container = document.createElement('div');
 
   componentDidMount() {
     this.addPortal();
 
-    $(document).on(`turbolinks:before-cache.${this.eventId}`, this.removePortal);
+    $(document).on('turbolinks:before-cache', this.removePortal);
   }
 
-  componentWillUnmount = () => {
+  componentWillUnmount() {
     this.removePortal();
 
-    $(document).off(`turbolinks:before-cache.${this.eventId}`);
-  };
+    $(document).off('turbolinks:before-cache', this.removePortal);
+  }
 
-  removePortal = () => {
-    if (this.container.parentElement === document.body) {
-      document.body.removeChild(this.container);
-    }
-  };
+  render() {
+    return createPortal(this.props.children, this.container);
+  }
 
-  render = () => createPortal(this.props.children, this.container);
+  private readonly addPortal = () => document.body.appendChild(this.container);
```

Comment:
Can use update to `window.newBody` otherwise it mounts to the previous version instead of the rendered one when navigating backwards